beater/beater.go

```diff
@@ -45,19 +43,17 @@ func (bt *ApmServer) Run(b *beat.Beat) error {
 		go bt.client.PublishAll(events)
 	}
 
-	bt.server, err = server.New(bt.config.Server)
-	if err != nil {
-		return err
-	}
-	bt.server.Start(callback, bt.config.Host)
-	defer bt.server.Stop()
+	server := apmServer(bt.config, callback)
+	start(server, bt.config.SSL)
```

Comment:
> It seems to me start and stop methods on the server itself are very natural.

I agree, but there isn't such server anymore, other than the builtin. I'm simply removing as much unneeded code as I can.

There is only 1 config, there is already 1 struct (`beater`) holding that config and passing it around, it doesn't make sense to have yet another struct (`server`) holding the same config to pass it around again.

Before it could make sense, because there was 2 configs, and there was 1 struct for each (among other stuff). It was also a source of confusion as I was reading things like `server.config`, `config.server`, `config.config` etc. 
Now the only `server` is golang builtin server, and the only config is the `beater`.